File: Queue/src/Queue.java
import java.util.LinkedList;
import java.util.NoSuchElementException;

public class Queue {
    private LinkedList<Integer> items;

    public Queue(){
        this.items = new LinkedList<>();
    }

    public void enqueue(int ele){
        this.items.add(ele);
    }
    public int dequeue(){
        if(isEmpty()){
            throw new NoSuchElementException("Queue underflow");
        }
        return this.items.removeFirst();
    }
    public int front(){
        if(isEmpty()){
            throw new NoSuchElementException("Queue underflow");
        }
        return this.items.getFirst();
    }

    public boolean isEmpty(){
        return this.items.isEmpty();
    }
    public int size(){
        return this.items.size();
    }
    public void print(){
        for (int item : this.items){
            System.out.print(item + " ");
        }
        System.out.println();
    }

    public static void main(String[] args) {
        Queue queue = new Queue();
        queue.enqueue(1);
        queue.enqueue(2);
        queue.enqueue(3);
        queue.enqueue(4);
        queue.print();
        System.out.println(queue.dequeue());
        System.out.println(queue.front());
        System.out.println(queue.isEmpty());
        System.out.println(queue.size());
    }
}

File: Queue/src/queue.js
class Queue {
    constructor() {
        this.items = [];
    }
    enqueue(ele){
        this.items.push(ele);
    }
    dequeue(){
        if(this.isEmpty()){
            return "Underflow"
        }
        return this.items.shift();
    }

    front(){
        if(this.isEmpty()){
            return "Underflow"
        }
        return this.items[0];
    }
    isEmpty(){
        return this.items.length === 0;
    }
    size(){
        return this.items.length;
    }
    print(){
        console.log(this.items.join(' '));
    }
}

const queue = new Queue();
queue.enqueue(1);
queue.enqueue(2);
queue.enqueue(3);
queue.enqueue(4)
queue.print();
console.log(queue.dequeue());
console.log(queue.front());
console.log(queue.isEmpty());
console.log(queue.size());